workbench-core/datasets/src/dataSetService.ts

```diff
@@ -289,14 +295,15 @@ export class DataSetService {
 
       if (
         !targetDS.externalEndpoints ||
-        !_.find(targetDS.externalEndpoints, (ep) => ep === externalEndpointId)
-      )
+        !_.find(targetDS.externalEndpoints, (ep) => ep === targetEndpoint.name)
+      ) {
         return;
+      }
 
       await storageProvider.removeExternalEndpoint(targetEndpoint.name, targetDS.awsAccountId!);
 
       targetDS.externalEndpoints = _.remove(targetDS.externalEndpoints, (endpoint) => {
```

Comment:
this code looks like it was always wrong. `_.remove()` returns the elements _removed_ from the array, not the remaining elements.

Reply:
Interesting. I think we need to take another look at our unit/integ tests for datasets because I feel like this should have been caught by them.

Reply:
what was the verdict here? Are we updating the dataset with the endpoints removed instead of the endpoints remaining?

Reply:
I'll take a look in my next ticket to double check things are working as intended. Want to get this PR merged